packages/headless/src/app/commerce-engine/commerce-engine-configuration.ts

```diff
@@ -57,7 +57,6 @@ export function getSampleCommerceEngineConfiguration(): CommerceEngineConfigurat
       currency: 'USD',
```

Comment:
Hey, Louis recently added a function called `navigatorContext` that returns values for `userAgent`, `referrer` and `url`.
The benefit for using the function instead of relay is to allow implementers to pass values to headless in a server-side rendering (SSR) scenario. If we only use relay internally, it would not be possible to configure in that context,

Reply:
Should we not retrieve the clientId from relay in that case, as is done in https://github.com/coveo/ui-kit/pull/4111?

Reply:
That's a great point, indeed the same argument would apply there. The default `navigatorContext` used in the browser could return the different properties using relay, while still allowing one to specify them in an SSR scenario,

Reply:
@samisayegh , does this mean than in an SSR scenario (which is the default scenario we will ultimately want to push for, especially for commerce implementations), Relay will not be able to extract the client ID?

Reply:
That's correct, SSR is completely different to the browser. A `clientId` would either:
1. Read from the request header if present.
2. Need to be generated on the server and returned to the browser e.g., via a response header

The clientId needs to be given to headless in an SSR context

We can export functions from relay (or a different package) to help. It'll be better to design how it looks like first though.

Naively, 

```
app.get('/', (req, res) => {
  const clientId = getOrGenerateClientId(req.headers) // provided by relay?

  const engine = buildCommerceEngine({
    ...,
    navigatorContext: () => {
     clientId,
     userAgent: req.userAgent,
     etc..
    }
  })
  ...

  headlers = enrichHeadersWithClientId(headers) // provided by relay?
  res.headers(headers) 
})
```
If headless SSR accepts and returns headers, we could simplify things quite a bit 🤞